packages/google_maps_flutter/google_maps_flutter/android/src/main/java/io/flutter/plugins/googlemaps/GoogleMapController.java

```diff
@@ -880,7 +852,11 @@ private int getActivityHashCode() {
     if (registrar != null && registrar.activity() != null) {
```

Comment:
Mmmm staring at this snippet a little more I'm worried this may crash with the old embedding when the activity is in the background and some lifecycle event for the same application happens.
We used to cache the activityHashCode, and we stopped caching at https://github.com/flutter/plugins/pull/2488 cc @cyanglaz do you remember the context for this change? was it safe to stop caching the activity hashCode in that PR?